mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -107,25 +107,41 @@ private[ml] trait DecisionTreeParams extends PredictorParams
   setDefault(maxDepth -> 5, maxBins -> 32, minInstancesPerNode -> 1, minInfoGain -> 0.0,
     maxMemoryInMB -> 256, cacheNodeIds -> false, checkpointInterval -> 10)
 
-  /** @group setParam */
+  /**
+   * @deprecated This method is deprecated and will be removed in 2.2.0.
+   * @group setParam
+   */
+  @deprecated("This method is deprecated and will be removed in 2.2.0.", "2.1.0")
```

Comment:
This will not appear in estimator classes, only in models. This can be confirmed by the following ways:
The IDE will prompt deprecated methods only for models: 
![image](https://cloud.githubusercontent.com/assets/1962026/20699424/247abfc4-b5bc-11e6-8eb2-a9fe53661cba.png)
We can also confirm it in Scala API docs:
```DecisionTreeClassifier```
![image](https://cloud.githubusercontent.com/assets/1962026/20699474/63b03688-b5bc-11e6-9e8d-b6d83f6bb00a.png)
```DecisionTreeClassificationModel```
![image](https://cloud.githubusercontent.com/assets/1962026/20699522/a0bf3556-b5bc-11e6-9289-d2c893f9428e.png)